src/main/OP.ts

```diff
@@ -246,13 +246,9 @@ function createVerifyRequestOptsFromBuilderOrExistingOpts(opts: {
           checkLinkedDomain: opts.builder.checkLinkedDomain,
           resolveOpts: {
             subjectSyntaxTypesSupported: subjectSyntaxTypesSupported,
-            resolver: opts.builder.resolver,
-            resolvers:
-              //TODO: discuss this with Niels
-              //https://sphereon.atlassian.net/browse/VDX-139
-              opts.builder.resolvers && opts.builder.resolvers.size
-                ? opts.builder.resolvers
-                : getResolver({ subjectSyntaxTypesSupported: subjectSyntaxTypesSupported }),
+            resolver: opts.builder.customResolver
+              ? opts.builder.customResolver
+              : getResolverUnion(opts.builder.responseRegistration.subjectSyntaxTypesSupported, opts.builder.resolvers),
```

Comment:
I think we need to make some changes, as the handling is onconsistent. What needs to happen is that we change this method for instance:

````typescript
 addDidMethod(didMethod: string, opts?: { resolveUrl?: string; baseUrl?: string }): OPBuilder {
    if (didMethod.startsWith('did:')) {
      this.addResolver(getMethodFromDid(didMethod), new Resolver(getUniResolver(getMethodFromDid(didMethod), { ...opts })));
      this.didMethods.push(getMethodFromDid(didMethod));
    } else {
      this.addResolver(didMethod, new Resolver(getUniResolver(didMethod, { ...opts })));
      this.didMethods.push(didMethod);
    }
    return this;
  }
````
As you can see, if you now call addDidMethod it will always create a new uniresolver for it. Which is similar to what you are doing in the getResolverUnion. The easy fix would be to just allow the map to have a null/undefined resolvable as value in the resolvers property, so that you indeed can handle it in your code


Also addResolver. directly manipulates the subjectSyntaxTypes supported it seems, whilst addDidMethod first populates the 'didMethods' property, which then gets merged into the subjectSyntaxTypes when the buildder builds the OP. We need to do one, or the other, but not mix them

Reply:
removed the `didMethods` from the builder files. now we just populate the `Map<string, Resolvable>` and then later on, in the actual build of the OP/RP, we add the did values from the map to `subjectSyntaxTypes`.
But I also have an unrelated questions about this. what is the correct form of a did method? let me rephrase that, what do we want as the end result of our process?
1. `example`
2. `did:example`
3. `did:example:`

right now I have changed the end result to be `did:example` I'm asking because we have a method in `DidJWT` called `toSIOPRegistrationDidMethod` which produces `did:example:`

Reply:
As mentioned. The SIOPv2 spec incorrectly has did:example as a DID method (it was worse when they had did:example:  as DID method). The actual DID method is only the 'example' portion of it (see DID core spec). 
So as long as we make sure that we accept did:example, example or even did:example:1234567 then correct that to a correct DID method (example) internally. Then whenever we start populating a SIOP payload, we have to adjust to did:example. The toSIOPRegistrationDidMethod is meant for this (it adds the last colon, because that was in the spec back then)

Reply:
fixed this according to our talk